test/e2e-shared-tests/mcd_config_drift.go

```diff
@@ -218,6 +231,62 @@ func (c configDriftTest) Run(t *testing.T) {
 				})
 			},
 		},
+		{
+			name:           "update degraded reason on drift",
+			rebootExpected: false,
+			testFunc: func(t *testing.T) {
+				mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
+				assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)
+
+				node, err := c.ClientSet.CoreV1Interface.Nodes().Get(context.TODO(), c.node.Name, metav1.GetOptions{})
```

Comment:
Feel free to skip for now:

Create a cancelable context within the testFunc that is used instead of `context.TODO()`. Ideally, it should be canceled when the test execution finishes. To wire that up, you can do the following:

```golang
ctx, cancel := context.WithCancel(context.Background())
t.Cleanup(cancel)

mutateFileOnNode(t, c.ClientSet, c.node, configDriftFilename, "not-the-data")
assertNodeAndMCPIsDegraded(t, c.ClientSet, c.node, c.mcp, configDriftFilename)

node, err := c.ClientSet.CoreV1Interface.Nodes().Get(ctx, c.node.Name, metav1.GetOptions{})

// ...
```

Also, if you do that, don't worry about fixing all the other `context.TODO()` calls for now.

Reply:
ok cool, I didn't know about the custom contexts, I'll wire that up.

Reply:
These tests don't really need them since they usually run in a single Goroutine. It's a good habit to get into though whenever you need a context.